tests/test_verify_base_url.py

```diff
@@ -54,3 +60,21 @@ def test_disable_verify_via_env(testdir, httpserver, monkeypatch):
     httpserver.serve_content(content='<h1>Error!</h1>', code=500)
     result = testdir.runpytest('--base-url', httpserver.url)
     assert result.ret == 0
+
+
+def test_verify_retries_url(testdir, httpserver, monkeypatch):
```

Comment:
Sorry, I don't understand the question. To check that it failed before recovering, we could try to check for the the line in stderr that includes the 500 response as mentioned in my previous comment.